src/MongoDB/ServerDescription.c

```diff
@@ -0,0 +1,268 @@
+
+	intern = Z_SERVERDESCRIPTION_OBJ_P(getThis());
+
+	PHONGO_PARSE_PARAMETERS_NONE();
+
+	RETVAL_LONG(mongoc_server_description_round_trip_time(intern->server_description));
+} /* }}} */
+
+/* {{{ proto integer MongoDB\Driver\ServerDescription::getType()
+   Returns the server’s node type */
+static PHP_METHOD(ServerDescription, getType)
+{
+	php_phongo_serverdescription_t* intern;
+
+	intern = Z_SERVERDESCRIPTION_OBJ_P(getThis());
+
+	PHONGO_PARSE_PARAMETERS_NONE();
+
+	RETVAL_LONG(php_phongo_server_description_type(intern->server_description));
+} /* }}} */
+
+/* {{{ MongoDB\Driver\ServerDescription function entries */
+ZEND_BEGIN_ARG_INFO_EX(ai_ServerDescription_void, 0, 0, 0)
+ZEND_END_ARG_INFO()
+
+static zend_function_entry php_phongo_serverdescription_me[] = {
+	/* clang-format off */
+	PHP_ME(ServerDescription, getHelloResponse, ai_ServerDescription_void, ZEND_ACC_PUBLIC | ZEND_ACC_FINAL)
+	PHP_ME(ServerDescription, getHost, ai_ServerDescription_void, ZEND_ACC_PUBLIC | ZEND_ACC_FINAL)
+	PHP_ME(ServerDescription, getLastUpdateTime, ai_ServerDescription_void, ZEND_ACC_PUBLIC | ZEND_ACC_FINAL)
+	PHP_ME(ServerDescription, getPort, ai_ServerDescription_void, ZEND_ACC_PUBLIC | ZEND_ACC_FINAL)
+	PHP_ME(ServerDescription, getRoundTripTime, ai_ServerDescription_void, ZEND_ACC_PUBLIC | ZEND_ACC_FINAL)
+	PHP_ME(ServerDescription, getType, ai_ServerDescription_void, ZEND_ACC_PUBLIC | ZEND_ACC_FINAL)
+	ZEND_NAMED_ME(__construct, PHP_FN(MongoDB_disabled___construct), ai_ServerDescription_void, ZEND_ACC_PRIVATE | ZEND_ACC_FINAL)
+	PHP_FE_END
+	/* clang-format on */
+};
+/* }}} */
+
+/* {{{ MongoDB\Driver\ServerDescription object handlers */
+static zend_object_handlers php_phongo_handler_serverdescription;
+
+static void php_phongo_serverdescription_free_object(zend_object* object) /* {{{ */
+{
+	php_phongo_serverdescription_t* intern = Z_OBJ_SERVERDESCRIPTION(object);
+
+	zend_object_std_dtor(&intern->std);
+
+	if (intern->properties) {
+		zend_hash_destroy(intern->properties);
+		FREE_HASHTABLE(intern->properties);
+	}
+
+	if (intern->server_description) {
+		mongoc_server_description_destroy(intern->server_description);
+	}
+}
+
+static zend_object* php_phongo_serverdescription_create_object(zend_class_entry* class_type) /* {{{ */
+{
+	php_phongo_serverdescription_t* intern = NULL;
+
+	intern = PHONGO_ALLOC_OBJECT_T(php_phongo_serverdescription_t, class_type);
+
+	zend_object_std_init(&intern->std, class_type);
+	object_properties_init(&intern->std, class_type);
+
+	intern->std.handlers = &php_phongo_handler_serverdescription;
+
+	return &intern->std;
+} /* }}} */
+
+HashTable* php_phongo_serverdescription_get_properties_hash(phongo_compat_object_handler_type* object, bool is_debug) /* {{{ */
+{
+	php_phongo_serverdescription_t* intern = NULL;
+	HashTable*                      props;
+
+	intern = Z_OBJ_SERVERDESCRIPTION(PHONGO_COMPAT_GET_OBJ(object));
+
+	PHONGO_GET_PROPERTY_HASH_INIT_PROPS(is_debug, intern, props, 2);
```

Comment:
The final parameter here is the initial size of the HashTable. This should agree with the number of properties we expect to fill in below to avoid PHP having to dynamically resize the HashTable.

```suggestion
	PHONGO_GET_PROPERTY_HASH_INIT_PROPS(is_debug, intern, props, 6);
```